swarm/CHANGELOG.md

```diff
@@ -6,6 +6,11 @@
   See [PR 4225](https://github.com/libp2p/rust-libp2p/pull/4225).
 - Remove deprecated `keep_alive_timeout` in `OneShotHandlerConfig`.
   See [PR 4677](https://github.com/libp2p/rust-libp2p/pull/4677).
+- Simplify the connection keep-alive algorithm.
+  Connections are now kept alive by default if there are active `Stream`s on it.
+  As a result, the `ConnectionHandler::connection_keep_alive` method now has a default implementation that returns `false`.
```

Comment:
```suggestion
  In case of no active `Stream`s the `ConnectionHandler` is asked whether it wants to keep the connection alive via  `ConnectionHandler::connection_keep_alive` which defaults to `false`.
```

Would be easier for me to understand, more specifically that `connection_keep_alive` is only called in case there are no active streams.

Reply:
updated 92823e1